src/InterventionBackend.php

```
@@ -255,29 +255,61 @@ public function getImageResource()
         // Handle resource
         $error = self::FAILED_UNKNOWN;
         try {
-            // write the file to a local path so we can extract exif data if it exists.
-            // Currently exif data can only be read from file paths and not streams
-            $tempPath = $this->config()->get('local_temp_path') ?? TEMP_PATH;
-            $path = tempnam($tempPath, 'interventionimage_');
-            if ($extension = pathinfo($assetContainer->getFilename(), PATHINFO_EXTENSION)) {
-                //tmpnam creates a file, we should clean it up if we are changing the path name
-                unlink($path);
-                $path .= "." . $extension;
+            // default value if reading exif data fails
+            $orientation = false;
+
+            // Try to read exif data from stream resource, fail silently
+            // This needs to be done before `getImageManager()->make()` because
+            // somewhere along that process the stream changes and exif becomes
+            // unreadable
+            try {
+                $exif = @exif_read_data($stream);
```

We shouldn't be using `@` to silence errors